### PR TITLE
Throttle refactors

### PR DESCRIPTION
**Pull request by smarshall-spitzbart**
## Description

Accomplishes refactors (and one slight functionality change) that were requested in #462. Also adds some comments 

## Type of change

- [x] `Feature`: Changes and/or adds code behavior, irrelevant to bug fixes
- [x] `Refactor`: Changes existing code style, naming, structure, etc.
- [x] `Docs`: Adds documentation

## New behavior tests

`TestPanicIfTooMuchThrottledPacketData` was slightly changed, in that we now panic the provider when any queue reaches `MaxThrottledPackets`. Previous behavior is that we'd panic when the queue was strictly larger than the max value.

## Regression tests
Everything else should act as a regression test

### CODE REVIEW

File: x/ccv/provider/keeper/throttle_test.go [hunk 1]
```diff
@@ -895,7 +897,7 @@ func TestPanicIfTooMuchThrottledPacketData(t *testing.T) {
 
 		// Queue packet data instances until we reach the max (some slash packets, some VSC matured packets)
 		reachedMax := false
-		for i := 0; i < int(tc.max+2); i++ { // iterate up to tc.max+1
+		for i := 0; i < int(tc.max); i++ {
```
Comment (smarshall-spitzbart) [inline x/ccv/provider/keeper/throttle_test.go:900]: New iteration follows more common conventions, old iteration was kind of confusing

**Review by mpoke — APPROVED** [review]
LGTM

File: x/ccv/provider/types/params.go [hunk 2]
```diff
@@ -35,9 +35,9 @@ const (
 	// fraction of total voting power that the slash meter can hold.
 	DefaultSlashMeterReplenishFraction = "0.05"
 
-	// DefaultMaxPendingSlashPackets defines the default maximum amount of pending slash packets that can
-	// be queued for a consumer before the provider chain halts.
-	DefaultMaxPendingSlashPackets = 1000
+	// DefaultMaxThrottledPackets defines the default amount of throttled slash or vsc matured packets
+	// that can be queued for a single consumer before the provider chain halts.
+	DefaultMaxThrottledPackets = 1000
```
Comment (MSalopek) [inline x/ccv/provider/types/params.go:40]: This is an arbitrary number, right?

I'm a bit uneasy about halting the provider if 1 consumer exceeds this number.

I'm fine with having this in, provided we treat consumers with a lot less trust in the coming iterations.

Could we devise a different approach where we would remove a consumer if this ever happens? If a removal event happens the consumer chain should be able to submit a governance proposal to be reinstated in ICS.

File: x/ccv/provider/types/keys.go [hunk 3]
```diff
@@ -106,7 +106,7 @@ const (
 	// ThrottledPacketDataBytePrefix is the byte prefix storing throttled packet data
 	ThrottledPacketDataBytePrefix
 
-	// PendingSlashPacketEntryBytePrefix is the byte prefix storing global slash queue entries
+	// GlobalSlashEntryBytePrefix is the byte prefix storing global slash queue entries
```
Comment (sainoe) [inline x/ccv/provider/types/keys.go:109]: Using this change to point out that the `SlashMeterByteKey` and  `LastSlashMeterFullTimeByteKey` constants above should have their name and doc updated respectively to sthg like:
```
// SlashMeterBytePrefix ...
-SlashMeterByteKey
+SlashMeterBytePrefix

----

-// LastSlashMeterReplenishTimeBytePrefix ...
+ // LastSlashMeterFullTimeByteKey ...
LastSlashMeterFullTimeByteKey
```

**Review by sainoe — APPROVED** [review]
LGTM!  Just nits from a previous Slash Throttle commit

File: x/ccv/provider/types/params.go (hunk 2 above)
Comment (smarshall-spitzbart) [reply x/ccv/provider/types/params.go:40]: I'd prefer to remove the halt code altogether once we have better mechanisms to throw out spam slash packets. Removing the consumer in the event that too many slash packets are recv seems like going too far down the route of safety mechanisms, similar to https://github.com/cosmos/interchain-security/issues/547

I'm going to continue merging this PR tho, as this conversation can be continued in open issues and the changes don't directly relate to halting the provider

File: x/ccv/provider/types/keys.go (hunk 3 above)
Comment (smarshall-spitzbart) [reply x/ccv/provider/types/keys.go:109]: The comments were off, thanks!

**Pull request merged**